File: javascript/test_sync.js
#!/usr/bin/env node
// test_sync.py — portiert nach javascript
// Quelle: python, OpenClaw@gateway1:scripts/test_sync.py
// auch in: OpenClaw@gateway2:scripts/test_sync.py
// Erzeugt: 2026-08-22 durch ABSTRACTIONS_MANAGER.py

// Test für Sync-Script

const fs = require('fs');
const path = require('path');

// Add the script path to module search paths
const scriptPath = '/home/openclaw/.openclaw/workspace/scripts';
const syncModulePath = path.join(scriptPath, 'sync_clawhub_git.js');

// Dynamically import the sync module
const { syncToGit, log } = require(syncModulePath);

// Test: db-maintainer ClawHub → Git (DRY-RUN)
console.log("=== TEST: db-maintainer sync (DRY-RUN) ===");
const skill = "db-maintainer";
const result = syncToGit(skill, true);
console.log(`Result: ${result ? 'SUCCESS' : 'FAILED'}`);

console.log("\n=== LOG-Inhalt ===");
try {
    const logContent = fs.readFileSync("/home/openclaw/.openclaw/workspace/logs/sync.log", "utf8");
    console.log(logContent);
} catch (error) {
    console.error("Fehler beim Lesen der Log-Datei:", error.message);
}

File: perl5/test_sync.pl
#!/usr/bin/perl
# test_sync.py — portiert nach perl5
# Quelle: python, OpenClaw@gateway1:scripts/test_sync.py
# auch in: OpenClaw@gateway2:scripts/test_sync.py
# Erzeugt: 2026-08-22 durch ABSTRACTIONS_MANAGER.py

use strict;
use warnings;

# Test für Sync-Script

use lib '/home/openclaw/.openclaw/workspace/scripts';
require 'sync_clawhub_git.pl';

# Test: db-maintainer ClawHub → Git (DRY-RUN)
print "=== TEST: db-maintainer sync (DRY-RUN) ===\n";
my $skill = "db-maintainer";
my $result = sync_to_git($skill, 1); # dry_run = true
print "Result: " . ($result ? 'SUCCESS' : 'FAILED') . "\n";
print "\n=== LOG-Inhalt ===\n";
open(my $fh, '<', '/home/openclaw/.openclaw/workspace/logs/sync.log') or die "Kann Log-Datei nicht öffnen: $!";
while (my $line = <$fh>) {
    print $line;
}
close($fh);
